### PR TITLE
feat(components): [table] support diagonal style

### PR DESCRIPTION
**Pull request by ccj-007**
In some categorized tables, the upper left corner cells need to be separated by diagonal lines, which we can achieve directly through diagonal and achieve copywriting display through header's slot.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 704186f</samp>

This pull request adds a new feature to the `table-column` component, which allows rendering the column header text diagonally. The feature is implemented by adding a new `diagonal` prop, updating the documentation and examples, adding a unit test, and modifying the style helper and the CSS file.

The table supports display diagonal style in the upper left corner

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at 704186f</samp>

*  Add a new attribute `diagonal` to the table-column component, which allows showing a slash style in the upper left corner of the table ([link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-48aed1abfbd975e78df7b5352f8c1321bb18c360349e47559acde0c4346b7123R355), [link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-5099e4c32f9d4d15337ce54f445e0afd4e558934baebe9387640c28f91a86dc3R67), [link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-5099e4c32f9d4d15337ce54f445e0afd4e558934baebe9387640c28f91a86dc3R149-R152), [link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-ace64c155a215e52adf697d6f56a3f5a077376cb59f2c725be070ce427a043e9R106), [link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-26d9963f276e146ddcfcd853509becc00b394ee0d1853c90f00d68546141a743R84-R87), [link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-50fcce7b86d8998ca0385dda1351e7376e5fa09b4939adec02eb34dda9e32ba5R209-R218))
*  Modify the custom-header example in `docs/examples/table/custom-header.vue` to demonstrate the usage of the `diagonal` attribute ([link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-745d3a71a368593a7524738ac360b8f5440fd718902e63de204932f6604dfd2bL3-R10))
*  Add a unit test for the `diagonal` attribute in `packages/components/table/__tests__/table-column.test.ts`, which checks that the first column header has the `is-diagonal` class when the attribute is set to true ([link](https://github.com/element-plus/element-plus/pull/13517/files?diff=unified&w=0#diff-ec759e2927b9c6eeb66ceb6fed5d7c5a179e0e1708c120d85406d94a51917d1bR583-R592))

In some classified tables, the upper left corner cells need to be separated by diagonal lines, which we can implement directly through diagonal and copy through header's slot

### CODE REVIEW

File: packages/components/table/src/table-column/defaults.ts [hunk 1]
```diff
@@ -145,4 +146,8 @@ export default {
       )
     },
   },
+  diagonal: {
+    type: Boolean,
+    default: false,
```
Comment (btea) [inline packages/components/table/src/table-column/defaults.ts:151]: The default value may be removed, refer to the [link](https://vuejs.org/guide/components/props.html#prop-validation). `The Boolean absent props will be cast to false`.

Comment (ccj-007) [reply packages/components/table/src/table-column/defaults.ts]: Thank you, the default value have been removed

**Review by btea — APPROVED** [review]
LGTM

**Comment by ccj-007** [comment]
This is my first time contributing code to element-plus, This has not been merged for a long time. Did I miss something @btea

**Comment by btea** [comment]
> This is my first time contributing code to element-plus, This has not been merged for a long time. Did I miss something @btea

Don't worry, there's no problem. We also need other members to review.

**Comment by tolking** [comment]
Since you still need to customize the `header ` slot after config this `diagonal` attribute, why not directly customize it?

```html
<el-table-column label="Date" prop="date" className="is-diagonal">
      <template #header>
        <div style="display: flex; justify-content: space-between">
          <div>Date</div>
          <div>Prop</div>
        </div>
      </template>
    </el-table-column>
```

**Comment by ccj-007** [comment]
Diagonal in el-table-column represents only the application style, using the original header slots for better customization. I would prefer developers to have more autonomy.  @tolking

**Comment by tolking** [comment]
IMO, If only the default style is added, users can easily implement it themselves. This `diagonal` attribute is not necessary.

Since you are implementing diagonals based on linear-gradient, it will also bring some additional issues

1. Columns of different sizes have different widths
2. Conflicting with group header style
3. There may also be slight differences in color from the border

**Comment by tolking** [comment]
Thank for your contribution.

**Pull request closed without merging**